### PR TITLE
Enable qpu.backend for IonQ backends

### PR DESCRIPTION
**Pull request by splch**
use the new `backend` field:

```python
dev = qml.device(
    'ionq.qpu',
    backend='aria-1',
    api_key=api_key,
    wires=2,
)
```

to call `qpu.aria-1`

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -11,11 +11,17 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import os
```
Comment (lillian542) [inline tests/conftest.py:14]: This import is needed in line 121

File: pennylane_ionq/device.py [hunk 2]
```diff
@@ -298,3 +327,98 @@ def generate_samples(self):
         samples = np.repeat(np.arange(number_of_states), counts)
         np.random.shuffle(samples)
         return QubitDevice.states_to_binary(samples, self.num_wires)
+
+
+# Specific Backends
```
Comment (lillian542) [inline pennylane_ionq/device.py:332]: Let's stick to `qml.device(“ionq.qpu”, backend=“backend-name”, api_key=api_key, wires=2)` and avoid adding specific subclasses for each hardware unit

File: README.rst [hunk 3]
```diff
@@ -108,7 +108,8 @@ You can instantiate the IonQ devices for PennyLane as follows:
 
     import pennylane as qml
     dev1 = qml.device('ionq.simulator', wires=2, shots=1000)
-    dev2 = qml.device('ionq.qpu', wires=2, shots=1000)
+    dev2 = qml.device('ionq.qpu.harmony', wires=2, shots=1000)
+    dev3 = qml.device('ionq.qpu', backend='aria-1', wires=2, shots=1000)
```
Comment (lillian542) [inline README.rst:112]: The documentation in the readme is updated here (pending modification to remove the device-specific subclassing), but the plugin documentation needs a similar update. The `docs/devices.rst` file needs to be updated to show the new `backend` functionality, and to refer users to IonQ’s website to find out what backends are available.

File: tests/test_device.py [hunk 4]
```diff
@@ -246,7 +245,7 @@ def mock_submit_job(*args):
             pass
```
Comment (lillian542) [inline tests/test_device.py:245]: In `TestDevice` above, please add a test explicitly checking the initialization when specifying a `backend` kwarg, as well as the default behaviour when “backend=None”

**Review by lillian542 — COMMENTED** [review]
Thank you for your contribution to the plugin, this looks great! 🎉 

I would like to request one change in the implementation - I’m hesitant to add hardware-specific subclasses for different machines when there is no functional difference that needs to be supported. The call signature

```
qml.device(“ionq.qpu”, backend=“aria-1”, api_key=api_key, wires=2)
```

works well, and it won’t require a new version of the plugin to be released if IonQ adds, removes, or changes the names of any of the devices. This is generally also what we’ve done with other plugins - for examples, see the [Honeywell](https://docs.pennylane.ai/projects/honeywell/en/latest/devices.html) or [Rigetti](https://docs.pennylane.ai/projects/rigetti/en/latest/devices/qpu.html) plugin documentation. 

Other than that, there are just a couple of small things that need to be added for tests and documentation :)

File: pennylane_ionq/device.py (hunk 2 above)
Comment (splch) [reply pennylane_ionq/device.py:332]: alright that sounds good! itll definitely be easier to maintain this way :)

**Comment by lillian542** [comment]
Thanks! Can you also run `black -l 100 pennylane_ionq/` to update whatever the formatting check is bothered about?

The CodeCov "reduced coverage" looks to me like it's an artifact of reformatting and not a change in how much of the plugin functionality is covered by the tests.

**Review by lillian542 — APPROVED** [review]
Looks good, thanks!

**Pull request merged**